data_safe_haven/infrastructure/project_manager.py

```diff
@@ -288,10 +288,17 @@ def destroy(self) -> None:
     def ensure_config(self, name: str, value: str, *, secret: bool) -> None:
         """Ensure that config values have been set, setting them if they do not exist"""
         try:
-            self.stack.get_config(name)
+            existing_value = self.stack.get_config(name)
         except automation.CommandError:
             self.set_config(name, value, secret=secret)
 
+        if existing_value != value:
+            msg = (
+                f"Unchangeable configuration variable '{name}' not consistent, "
+                f"your configuration: '{value}', Pulumi workspace: '{existing_value}'."
+            )
+            raise DataSafeHavenPulumiError(msg)
+
```

Comment:
Not certain yet. However, none of those values would get updated if they were changed so if this change doesn't work I think it would mean there is a problem elsewhere.

Reply:
The options with `replace==False` are

- location
- subscription id
- tenant id

To change any of those would require many resources to be destroyed then newly created in the new sub/location/tenant.
I think it is safe to raise an exception in those cases as the most sensible thing to do would be to teardown and deploy a new SRE.

Reply:
OK. Is it worth renaming the function and/or updating the docstring to better indicate that these are immutable values?

Reply:
I'm happy with the name, the docstring/comments could do with updating though.

Reply:
Done in b16256f